tests/test_datamodel_api.py

```diff
@@ -292,14 +292,16 @@ def cb(val):
     service.set_state(app_name, "/A/X", "xyz")
     timeout_loop(lambda: called == 2, timeout=5)
     assert called == 2
-    # TODO: value is still "cde" in both old and new API
-    # assert value == "xyz"
+    test_name = request.node.name
+    # TODO: the value is not modiefied in the old API - issue
```

Comment:
This issue is not present in the new API - probably doesn't worth investigating for the old API.